### PR TITLE
Update TESTING.md 'Building with extra plugins' documentation

### PR DESCRIPTION
**Pull request by finnegancarroll**
### Description
["Building with extra plugins"](
https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#building-with-extra-plugins
) section of TESTING.md is stale. Updating docs to reflect plugin documentation.

### Related Issues
#15847

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -552,11 +552,17 @@ Apart from using Gradle, it is also possible to gain insight in code coverage us
 
 Please read your IDE documentation for how to attach a debugger to a JVM process.
 
-# Building with extra plugins
+# Testing with plugins
```
Comment (dblock) [inline TESTING.md:555]: Update TOC at the top of this document too, please.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good, just needs a TOC update. Thanks for doing this!

File: TESTING.md (hunk 1 above)
Comment (finnegancarroll) [reply TESTING.md:555]: Missed that, thanks!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
> ❌ Gradle check result for [7e2aaa0](https://github.com/opensearch-project/OpenSearch/commit/7e2aaa01b03a01e07e47fcfe08ad8d6a9b8e0b7b): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/47775/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

It's a markdown change, apologies if I was not supposed to merge without retrying gradle check here.

**Pull request merged**